pkg/plugin/validation.go

```diff
@@ -17,32 +17,64 @@ limitations under the License.
 package plugin
 
 import (
-	"errors"
 	"fmt"
 
 	"github.com/blang/semver"
 
 	"sigs.k8s.io/kubebuilder/pkg/internal/validation"
 )
 
+// errInvalidVersion should be returned when a plugin's version is invalid.
+type errInvalidVersion struct {
+	version, msg string
+}
+
```

Comment:
also, I'd suggest keep the struct definition on top and the methods on the bottom.